internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -443,24 +450,32 @@ func (c *enrollCmd) prepareFleetTLS() error {
 
 func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 	err := c.daemonReload(ctx)
+	if err != nil &&
```

Comment:
Why do we need to special case this? Why can't we just enter the backoff loop below and exit it once it is finished?